### PR TITLE
Implement DECIMAL to JSON casts

### PR DESCRIPTION
**Pull request by arhimondr**
@martint Another part of the #4958

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java [hunk 1]
```diff
@@ -123,12 +124,12 @@ private boolean matchesParameterAndReturnTypes(Method method, List<Type> resolve
 
         Class<?>[] methodParameterJavaTypes = method.getParameterTypes();
         for (int i = 0; i < resolvedTypes.size(); ++i) {
-            if (!methodParameterJavaTypes[i].equals(resolvedTypes.get(i).getJavaType())) {
+            if (!unwrap(methodParameterJavaTypes[i]).equals(unwrap(resolvedTypes.get(i).getJavaType()))) {
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java]: Why do we need this?

Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java]: Also, if this is a pre-existing issue, can you pull it out into a separate commit?

File: presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java [hunk 2]
```diff
@@ -86,6 +100,13 @@ private static SqlScalarFunction castFunctionFromDecimalTo(String to, String...
 
     private static SqlScalarFunction castFunctionToDecimalFrom(String from, String... methodNames)
     {
+        SqlScalarFunctionBuilder sqlScalarFunctionBuilder = castFunctionToDecimalFromBuilder(from, methodNames);
+        return sqlScalarFunctionBuilder
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/type/DecimalCasts.java]: Inline this variable

**Comment by arhimondr** [comment]
@martint Your comments have been addressed. Could you please merge this?

**Comment by martint** [comment]
The build is currently broken (due to unrelated changes). I'll merge this once we figure out what's going on.

**Comment by martint** [comment]
Merged, thanks!
